utils.py

```diff
@@ -269,6 +273,7 @@ def update_article_metadata(reader, folder_path=None, owner=None, import_id=None
                         ),
                     )
                 article = update_article(article, issue, prepared_row, folder_path)
+                articles.append(article)
```

Comment:
I don't think we need anything other than the article id, right? (`article.id)` I'm concerned about memory usage when loading an import for a large batch of articles since storing them into a list will essentially load the entire object in memory for not much purpose.

An alternative approach would be to turn the `actions` link into a dictionary mapping from article_id to list of actions, which saves us from returning redundant data

Reply:
Yes, I could do that. It will mean rewriting anything that actions returns to. Is that worth the time?

Reply:
If you don't want to touch the actions, then returning a separate list of article_ids is fine. it will have the same memory footprint. What we want to avoid is collecting all the objects in memory

Reply:
OK, I went ahead and based everything on article ids as the keys inside actions as a dictionary.